src/pages/AddItem/index.js

```diff
@@ -20,27 +24,39 @@ export const AddItemPage = () => {
   const onSubmit = (event) => {
     event.preventDefault();
     setIsLoading(true);
-
-    addProduct(formState)
-      .then((res) => console.log(res))
-      .catch(() => setError(true))
-      .finally(() => {
-        setIsLoading(false);
-
-        setFormState({
-          productName: '',
-          timeFrame: '7',
-          lastPurchaseDate: null,
+    /*
+    Checks the state if the product already exists
+    Find returns undefined if it has not matches
+    */
+    const normalizer = (input) => input.toUpperCase().replaceAll(/[.,:;]/g, '');
+    const normalizedInput = normalizer(formState.productName);
+    const checkProduct = products.find(
+      (product) => normalizer(product.productName) === normalizedInput,
+    );
+    /*
+    We have to compare explicitely if it is undefined, otherwise
+    it wouldn't work as expected
+    */
+    if (checkProduct === undefined) {
+      addProduct(formState)
+        .then((res) => console.log(res))
```

Comment:
```suggestion
        .then((res) => setMessage("'Successfully created product!'"))
```

I believe it's much better than `console.log`. 😄 What do you think?